### ISSUE TITLE
XLM-RoBERTa MLM Trainer not saving 'sentencepiece.bpe.model' file

### ISSUE THREAD

**Issue by ga-tardochisalles** [huggingface/transformers#12197]
## Environment info (Colab)

- `transformers` version: 4.7.0.dev0
- Platform: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.10
- PyTorch version (GPU?): 1.8.1+cu101 (True)
- Tensorflow version (GPU?): 2.5.0 (True)
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

## Information

Model I am using xlm-roberta-base:

The problem arises when using:
* [x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

Example: [RoBERTa/BERT/DistilBERT and masked language modeling, using HuggingFace Trainer with your own train file](https://github.com/huggingface/transformers/tree/master/examples/pytorch/language-modeling#robertabertdistilbert-and-masked-language-modeling)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

Dataset: csv file with only one column named "text", containing one sentence per row.

## To reproduce

1. Follow the instructions displayed in this [pytorch language-modeling examples](https://github.com/huggingface/transformers/tree/master/examples/pytorch/language-modeling#robertabertdistilbert-and-masked-language-modeling) page (RoBERTa/BERT/DistilBERT and masked language modeling, using HuggingFace Trainer with your own train file).

Used command: 
`!python train.py --model_name_or_path xlm-roberta-base --train_file custom_train_dset.csv --save_steps 300000 --line_by_line --do_train --output_dir xlm-roberta-base-mlm-tuned-example`

## Expected behavior

I did the very same thing, but with less data(same custom dataset, less rows) two days ago(2021/06/14) and I got the desired output:
![image](https://user-images.githubusercontent.com/60230715/122219560-f0bfb580-ce85-11eb-99a7-982c474d8050.png)

Now, this is the output that I am getting(**wrong**):
![image](https://user-images.githubusercontent.com/60230715/122218776-287a2d80-ce85-11eb-8878-622f58f8fb39.png)

@sgugger

**Comment by sgugger** [comment]
Without seeing your training script, it's impossible to diagnose what went wrong. I just tried a `tokenizer.save_pretrained(...)` with this model and I get all the files.

**Comment by ga-tardochisalles** [comment]
Hey @sgugger, thanks for the quick reply! I was making a very stupid mistake(typo) and haven't noticed it until now.
I was using 'roberta-base' instead of 'xlm-roberta-base', that is why there was no 'sentencepiece.bpe.model' file when saving it.
Sorry for taking your time!
